lib/analytics.js

```diff
@@ -78,6 +88,12 @@ module.exports = class Analytics {
     }
     const sessionId = await this._sessionIdProducer(request)
     const attribution = (await this._attributionProducer(request)) || {}
+    if (Object.keys(attribution).length > 0) {
+      const validationResult = attributionSchema.validate(attribution)
+      if (validationResult.error) {
+        throw new Error('Attribution should contain campaign and one of source or id')
```

Comment:
I'm not particularly comfortable with the plugin throwing exceptions which could impact the functionality of the services using it (for what is essentially non-critical functionality.) 

Could we just use a `console.warn` and accept that Analytics will not record the attribution correctly?